### PR TITLE
fixed portal code validation and agreement consent update

### PR DESCRIPTION
**Pull request by nidhigarg-bmw**

### CODE REVIEW

**Review by lavanya-bmw — APPROVED** [review]
lgtm

**Review by evegufy — CHANGES_REQUESTED** [review]
@nidhigarg-bmw this PR is towards main, please to to edit and change to dev:
image

File: src/components/cax-companyRole.tsx [hunk 1]
```diff
@@ -184,13 +184,22 @@ export const CompanyRoleCax = ({
...
+                      </p>
+                    )
+                  }else{
+                    return (
+                      <p className="agreement-text">
+                        <span>{agreement.name}</span>
+                      </p>
+                    )
+                  }
+
```
Comment (oyo) [inline src/components/cax-companyRole.tsx:202]: Better write this as
`return (`
`<p className="agreement-text">`
`{ agreement.documentIds.length`
`? <>{/*Line 190-192*/}</>`
`: <span>{agreement.name}</span>`
`})`

**Review by oyo — CHANGES_REQUESTED** [review]
slight simplification

**Review by evegufy — DISMISSED** [review]

**Pull request merged**